modules/platforms/dotnet/Apache.Ignite/Compute/ICompute.cs

```diff
@@ -15,60 +15,81 @@
  * limitations under the License.
  */
 
-namespace Apache.Ignite.Compute
-{
-    using System.Collections.Generic;
-    using System.Threading.Tasks;
-    using Network;
-    using Table;
+namespace Apache.Ignite.Compute;
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Network;
+using Table;
 
+/// <summary>
+/// Ignite Compute API provides distributed job execution functionality.
+/// </summary>
+public interface ICompute
+{
     /// <summary>
-    /// Ignite Compute API provides distributed job execution functionality.
+    /// Executes a compute job represented by the given class on one of the specified nodes.
     /// </summary>
-    public interface ICompute
-    {
-        /// <summary>
-        /// Executes a compute job represented by the given class on one of the specified nodes.
-        /// </summary>
-        /// <param name="nodes">Nodes to use for the job execution.</param>
-        /// <param name="jobClassName">Java class name of the job to execute.</param>
-        /// <param name="args">Job arguments.</param>
-        /// <typeparam name="T">Job result type.</typeparam>
-        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
-        Task<T> ExecuteAsync<T>(IEnumerable<IClusterNode> nodes, string jobClassName, params object?[]? args);
+    /// <param name="nodes">Nodes to use for the job execution.</param>
+    /// <param name="units">Deployment units.</param>
```

Comment:
Can we provide an empty list? Let's add these details to comments.

Reply:
Added "can be empty" note. Ideally we should explain what that means, but I don't have deep enough understanding of https://cwiki.apache.org/confluence/display/IGNITE/IEP-103%3A+Code+Deployment to do that.